cts-java/src/test/java/org/ga4gh/cts/api/reads/ReadGroupSetsPagingIT.java

```diff
@@ -0,0 +1,136 @@
+package org.ga4gh.cts.api.reads;
+
+import org.apache.avro.AvroRemoteException;
+import org.ga4gh.ctk.transport.URLMAPPING;
+import org.ga4gh.ctk.transport.protocols.Client;
+import org.ga4gh.cts.api.TestData;
+import org.ga4gh.cts.api.Utils;
+import org.ga4gh.methods.GAException;
+import org.ga4gh.methods.SearchReadGroupSetsRequest;
+import org.ga4gh.methods.SearchReadGroupSetsResponse;
+import org.ga4gh.models.ReadAlignment;
+import org.ga4gh.models.ReadGroupSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test the <tt>/readgroupsets/search</tt> paging behavior.
+ *
+ * @author Herb Jellinek
+ */
+@Category(ReadsTests.class)
+public class ReadGroupSetsPagingIT {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Check that we can page 1 by 1 through the {@link ReadGroupSet}s
+     * we receive from
+     * {@link Client.Reads#searchReadGroupSets(SearchReadGroupSetsRequest)}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingOneByOneThroughReadGroupSets() throws AvroRemoteException {
+
+        // retrieve them all - this may return fewer than "all."
+        final List<ReadGroupSet> listOfReadGroupSets = Utils.getAllReadGroupSets(client);
+        assertThat(listOfReadGroupSets).isNotEmpty();
+
+        // we will do a set comparison after retrieving them 1 at a time
+        final Set<ReadGroupSet> setOfExpectedReadGroupSets = new HashSet<>(listOfReadGroupSets);
+        assertThat(listOfReadGroupSets).hasSize(setOfExpectedReadGroupSets.size());
+
+        final Set<ReadGroupSet> setOfReadGroupSetsGathered1By1 =
+                new HashSet<>(setOfExpectedReadGroupSets.size());
+        // page through the ReadGroupSets using the same query parameters
+        String pageToken = null;
+        do {
+            final SearchReadGroupSetsRequest pageReq =
+                    SearchReadGroupSetsRequest.newBuilder()
+                                              .setDatasetId(TestData.getDatasetId())
+                                              .setPageSize(1)
+                                              .setPageToken(pageToken)
+                                              .build();
+            final SearchReadGroupSetsResponse pageResp = client.reads.searchReadGroupSets(pageReq);
+            final List<ReadGroupSet> pageOfReadGroupSets = pageResp.getReadGroupSets();
+            pageToken = pageResp.getNextPageToken();
+
+            assertThat(pageOfReadGroupSets).hasSize(1);
+            setOfReadGroupSetsGathered1By1.add(pageOfReadGroupSets.get(0));
+        } while (pageToken != null);
+
+        assertThat(setOfReadGroupSetsGathered1By1).containsAll(setOfExpectedReadGroupSets);
```

Comment:
I think this is something we should file as an issue to follow up on, but to my eye there's a lot of duplicated code testing the same functionality in terms of paging. I've made the point in the references tests as well. A simpler and more readable approach would be to make the page size an option in the client, and to provide methods that take the search method parameters directly (getting rid of the process of building requests entirely from these test cases). 

We should also have tests that provide malformed search parameters, of course. But for testing output of searching by name and so on, a method that returns a list/iterator would be a lot more useful than working with requests and doing paging every time.